p2panda-auth/src/lib.rs

```diff
@@ -1,5 +1,74 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
+//! `p2panda-auth` provides decentralised group management with fine-grained, per-member permissions.
+//!
+//! Once a group has been created, members can be added, removed, promoted and demoted. Each member
+//! has an associated access level which can be used to determine their permissions. The access
+//! levels are `Pull`, `Read`, `Write` and `Manage`. Each access level is a superset of the lower
+//! levels and can be assigned an associated set of conditions; this allows fine-grained
+//! partitioning of each access level. For example, `Read` conditions could be assigned with a path
+//! to restrict access to areas of a dataset. Finally, only members with `Manage` access are allowed
+//! to modify the group state by adding, removing, promoting or demoting other members.
+//!
+//! The access levels defined by `p2panda-auth` may prove useful when controlling replication of
+//! datasets. Custom sync protocols can be defined which rely on group membership and access levels
+//! to determine who to sync with and over which subsets of data.
```

Comment:
Could be nice to mention also how it helps building application-layer stuff (for example by establishing moderation rules around conditions and write access)